numpy/lib/arraysetops.py

```diff
@@ -261,6 +261,36 @@ def unique(ar, return_index=False, return_inverse=False,
         ret = _unique1d(ar, return_index, return_inverse, return_counts)
         return _unpack_tuple(ret)
 
+    if ar.size == 0:
+        # if there's no elements (at least one axis is 0), the reshaping and
+        # viewing doesn't work, but there's also no data, so those
+        # manipulations aren't needed to get right answer.
+
+        axis_len = ar.shape[axis]
```

Comment:
This won't raise `AxisError` appropriately.

Can you put these special cases just before `# Must reshape to a contiguous 2D` instead?

Reply:
Ah, nice catch; putting it after the `moveaxis` makes this a little more awkward (because it has to then also do `np.moveaxis` back to the right place), but not too bad.